src/utils/instance-utils.ts

```diff
@@ -0,0 +1,27 @@
+interface InstancesWithErrors<InstanceType> {
```

Comment:
Let's avoid random utils files whenever possible. In this case, this a function that works with Struct, so it can go to that file (as free function or static method)